### PR TITLE
Prepare 1.2.4 release

### PR DESCRIPTION
**Pull request by ElePT**
### Summary
Release preparation PR for 1.2.4. I re-reviewed the QPY fix reno to confirm that it never mentions 1.2.3.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1224,7 +1224,7 @@ dependencies = [
 
 [[package]]
 name = "qiskit-qasm3"
-version = "1.2.3"
+version = ""
```
Comment (ElePT) [inline Cargo.lock]:
```suggestion
version = "1.2.4"
```

File: releasenotes/notes/prepare-1.2.4-4a855c6e6bf78a4b.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+prelude: >
+    Qiskit 1.2.4 is a patch release that fixes a QPY problem introduced in the 
+    `yanked <https://peps.python.org/pep-0592/>`__ 1.2.3 bugfix release. In said
+    release, an error would be raised when loading a QPY file that was generated with 
+    a different symengine version from the version installed when loading.
```
Comment (mtreinish) [inline releasenotes/notes/prepare-1.2.4-4a855c6e6bf78a4b.yaml]: Maybe we can get @abbycross to take a quick look at this sentence.

```suggestion
    `yanked <https://peps.python.org/pep-0592/>`__ 1.2.3 bugfix release. In 1.2.3
    an error would be raised when loading a QPY file that was generated with 
    a different symengine version from the version installed when loading.
```

File: releasenotes/notes/prepare-1.2.4-4a855c6e6bf78a4b.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+prelude: >
+    Qiskit 1.2.4 is a patch release that fixes a QPY problem introduced in the 
+    `yanked <https://peps.python.org/pep-0592/>`__ 1.2.3 bugfix release. In 1.2.3
```
Comment (Eric-Arellano) [inline releasenotes/notes/prepare-1.2.4-4a855c6e6bf78a4b.yaml]:
```suggestion
    `yanked <https://peps.python.org/pep-0592/>`__ 1.2.3 bugfix release. In 1.2.3,
```

**Review by kevinhartman — APPROVED** [review]
LGTM.

**Pull request merged**